bundles/org.openhab.core.addon.marketplace/src/main/java/org/openhab/core/addon/marketplace/internal/json/JsonMarketplaceAddonService.java

```diff
@@ -0,0 +1,249 @@
+
+import org.openhab.core.addon.Addon;
+import org.openhab.core.addon.AddonEventFactory;
+import org.openhab.core.addon.AddonService;
+import org.openhab.core.addon.AddonType;
+import org.openhab.core.addon.marketplace.MarketplaceAddonHandler;
+import org.openhab.core.addon.marketplace.MarketplaceHandlerException;
+import org.openhab.core.config.core.ConfigurableService;
+import org.openhab.core.events.Event;
+import org.openhab.core.events.EventPublisher;
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.reflect.TypeToken;
+
+/**
+ * This class is a {@link AddonService} retrieving JSON marketplace information.
+ *
+ * @author Yannick Schaus - Initial contribution
+ * @author Jan N. Klug - Refactored for JSON marketplaces
+ */
+@Component(immediate = true, configurationPid = "org.openhab.core.marketplace.json", //
+        property = Constants.SERVICE_PID + "=org.openhab.core.marketplace.json")
+@ConfigurableService(category = "system", label = "JSON 3rd Party Marketplace", description_uri = JsonMarketplaceAddonService.CONFIG_URI)
+public class JsonMarketplaceAddonService implements AddonService {
+    static final String CONFIG_URI = "system:jsonmarketplace";
+    private static final String ADDON_ID_PREFIX = "jsonmarketplace:";
+
+    private static final Map<String, AddonType> TAG_ADDON_TYPE_MAP = Map.of( //
+            "automation", new AddonType("automation", "Automation"), //
+            "binding", new AddonType("binding", "Bindings"), //
+            "misc", new AddonType("misc", "Misc"), //
+            "persistence", new AddonType("persistence", "Persistence"), //
+            "transformation", new AddonType("transformation", "Transformations"), //
+            "ui", new AddonType("ui", "User Interfaces"), //
+            "voice", new AddonType("voice", "Voice"));
+
+    private final Gson gson = new GsonBuilder().setDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'").create();
+    private final Set<MarketplaceAddonHandler> addonHandlers = new HashSet<>();
+
+    private List<String> marketplaceUrls = List.of();
+    private List<AddonEntryDTO> cachedAddons = List.of();
+
+    private boolean showUnstable = false;
+
+    private EventPublisher eventPublisher;
+
+    @Activate
+    public void activate(Map<String, Object> config) {
+        modified(config);
+    }
+
+    @Modified
+    public void modified(Map<String, Object> config) {
+        String urls = Objects.requireNonNullElse((String) config.get("urls"), "");
+        marketplaceUrls = Arrays.asList(urls.split(","));
```

Comment:
We should use a different separator. A comma can be a valid separator in queries if an add-on provider implements a protocol like OData or similar. What about a pipe `|`?